### ISSUE TITLE
OCRmyPDF runs unpaper 2 times for each page if --clean-final is used

### ISSUE THREAD

**Issue by rbrito** [jbarlow83/OCRmyPDF#579]
**Describe the bug**

OCRmyPDF runs `unpaper` 2 times for each page if `--clean-final` is used (which I frequently use with some scans that I get--sometimes, even from well reputed publishers--I can mention at least one such case in private).

If I only use `--clean` (which is not always what I want), then it seems to only run `unpaper` once for each page.

**To Reproduce**

```bash
ocrmypdf --verbose 1 --keep-temporary-files --clean-final --tesseract-timeout=0 ccitt.pdf ccitt.ocr.pdf
```

The log showing that `unpaper` is run twice per page is attached. The output directory, in particular, contains 3 PNG files for the PDF above. They here is something interesting:

```
$ md5sum *.png
e07c289b05ce55162dee74cff67fa116  000001_ocr.png
e07c289b05ce55162dee74cff67fa116  000001_pp_clean.png
59e9915a4165e10651df1d64f9ad1153  000001_rasterize.png
```

I have always took a peek at the files when they are being generated and I have always seen that the `*_ocr.png` and the `*_pp_clean.png` are visually the same.

**Example file**

The "stock" `tests/resources/ccitt.pdf`, for instance.

**Expected behavior**

Keep the output of the first run of `unpaper` (since it should be deterministic, after all) and use it for the final PDF.

**System**
 - OS: Debian
 - OCRmyPDF Version:
```
(ocrmypdf) rbrito@zatz:/tmp$ ocrmypdf --version
10.0.1
(ocrmypdf) rbrito@zatz:/tmp$ 
```
 - How did you install ocrmypdf? `pip`, in a virtual env (version 10 of ocrmypdf is not yet in Debian).

[debug.log](https://github.com/jbarlow83/OCRmyPDF/files/4781496/debug.log)

**Comment by jbarlow83** [comment]
Fortunately, everything was working correctly - this was just an optimization opportunity not taken when `--clean-final` was set and `--remove-vectors` was not set.

**Comment by jbarlow83** [comment]
Fixed in v10.1.0
